examples/server-side-rendering/cypress/tests/spec.cy.js

```diff
@@ -0,0 +1,37 @@
+describe('template spec', () => {
+  it('passes with simple', () => {
+    cy.visit('/?entrypoint=simple', {
+    })
+    cy.window().then((win) => {
+      cy.spy(win, 'loadableEvents').as('spyloadableEvents')
+    })
+  
+
+    cy.get('@spyloadableEvents')
+      .should('be.calledWith', Cypress.sinon.match({ type: 'loadSync', chunkName: 'letters-A' }));
+  })
+  it('passes with props', () => {
+    cy.visit('/?entrypoint=props', {
+    })
+    cy.window().then((win) => {
+      cy.spy(win, 'loadableEvents').as('spyloadableEvents')
+    })
+  
+
+    cy.get('@spyloadableEvents')
+      .should('be.calledWith', Cypress.sinon.match({ type: 'loadSync', chunkName: 'letters-A' }))
+      .should('be.calledWith', Cypress.sinon.match({ type: 'loadSync', chunkName: 'letters-F' }));
+  })
```

Comment:
if you abstract this operation to a local helper you can make it shorter and get more "clear" signal as a result.

Good testing is all about removing _noise_ from the code in order to reveal intentions

Reply:
Not quite sure what that would look like. Got any pointers or ideas? 😀

Reply:
The general concept is perfectly described here - https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/ and https://glebbahmutov.com/blog/readable-tests/

About readability and _extraction of helpers_ to maximise signal from tests.
Cypress usually proposes custom commands, but I prefer plain functions as they are less "magical".

long story short
```tsx
// extract helper
const shouldObserve = (event) => {
   cy.get('@spyloadableEvents')
       .should('be.calledWith', Cypress.sinon.match(event))
}

// use helper

shouldObserve({ type: 'loadSync', chunkName: 'letters-A' });
shouldObserve({ type: 'loadSync', chunkName: 'letters-F' });
```

So there is almost no difference, but you are removing unnecessary details from the code and can focus on what matters more.